Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java

```diff
@@ -256,6 +262,20 @@ void setIsReqStartedFromBranchShareSheet(boolean startedByShareSheet) {
     boolean isReqStartedFromBranchShareSheet() {
         return isReqStartedFromBranchShareSheet_;
     }
+    
+    @Override
+    public JSONObject toJSON() {
+        JSONObject jsonObject = super.toJSON();
+        try {
+            jsonObject.putOpt(Defines.Jsonkey.CustomBranchLinkDataObj.getKey(), linkPost_.getLinkDataJsonObject());
```

Comment:
If the link data isn't getting persisted to preferences, shouldn't we just store the link data correctly rather than add a whole separate key to store the data in?

Reply:
Just read through the code to refresh myself. Seems like you need to store the other properties in here as well isReqStartedFromBranchShareSheet_, defaultToLongUrl_, isAsync_. Plus, there'll be no reference to the callback anymore after reloading which I imagine would be an issue, right?